src/adapters.ts

```diff
@@ -1644,31 +1838,123 @@ export function AdapterL1<TBase extends Constructor<TxSender>>(Base: TBase) {
+          'finalizeDeposit',
+          tx.data
+        );
+        const proof = await this._providerL2().getLogProof(
+          depositHash,
+          successL2ToL1LogIndex
+        );
+        if (!proof) {
+          throw new Error('Log proof not found!');
+        }
+        return await l1Bridge.claimFailedDeposit(
+          (await this._providerL2().getNetwork()).chainId as BigNumberish,
+          calldata['_l1Sender'],
+          calldata['_l1Token'],
+          calldata['_amount'],
+          depositHash,
+          receipt.l1BatchNumber!,
+          proof.id,
+          receipt.l1BatchTxIndex!,
+          proof.proof,
+          overrides ?? {}
+        );
+      } else {
+        const l1AR = IL1AssetRouter__factory.connect(
+          l1BridgeAddress,
+          this._signerL1()
+        );
+        const l2ARInterface = IAssetRouterBase__factory.createInterface();
+        const l1ntv = await this.getNativeTokenVaultL1();
+        try {
+          const calldata = l2Bridge.interface.decodeFunctionData(
+            'finalizeDeposit',
+            tx.data
+          );
 
-      const calldata = l2Bridge.interface.decodeFunctionData(
-        'finalizeDeposit',
-        tx.data
-      );
+          const proof = await this._providerL2().getLogProof(
+            depositHash,
+            successL2ToL1LogIndex
+          );
+          if (!proof) {
+            throw new Error('Log proof not found!');
+          }
 
-      const proof = await this._providerL2().getLogProof(
-        depositHash,
-        successL2ToL1LogIndex
-      );
-      if (!proof) {
-        throw new Error('Log proof not found!');
+          // todo: generally, providing token address is not required for the NTV
+          // this sdk does it, but we should support the case even when it is not given
+          const assetData = encodeNTVTransferData(
+            calldata['_amount'],
+            calldata['_l2Receiver'],
+            calldata['_l1Token']
+          );
+
+          const assetId = await l1ntv.assetId(calldata['_l1Token']);
+          if (assetId === ethers.ZeroHash) {
+            throw new Error(
+              `Token ${calldata['_l1Token']} not registered in NTV`
+            );
+          }
+
+          // todo: this SDK assumes that users used new encoding version,
+          // however in general it should be capable of working with hte old version as well
+          return await l1AR[
+            'bridgeRecoverFailedTransfer(uint256,address,bytes32,bytes,bytes32,uint256,uint256,uint16,bytes32[])'
+          ](
+            (await this._providerL2().getNetwork()).chainId as BigNumberish,
+            calldata['_l1Sender'], // depositSender
+            assetId, // asset id
+            assetData,
+            depositHash,
+            receipt.l1BatchNumber!,
+            proof.id,
+            receipt.l1BatchTxIndex!,
+            proof.proof,
+            overrides ?? {}
+          );
+        } catch (e) {
```

Comment:
Do we want to process every exception here like this?

Reply:
its for the decodeFunctionData, depending on the withdrawal method we need to withdraw differently

Reply:
I understand why it was added; the accept is on the word _every_. Shouldn't we check the type of exception?

Reply:
https://github.com/zksync-sdk/zksync-ethers/blob/4a75d296a7fb6a1cc1575e84f1159baf4e8a763f/src/provider.ts#L527

This is how it was done in main already. I can't debug this locally nicely, so I would just leave it as is